extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/devconsole/DevConsoleProcessor.java

```diff
@@ -69,23 +72,16 @@ public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs)
                     Path parentOfDir = dir.getParent().getFileName();
 
                     if (!staticResourceInfo.resourceMap.containsKey(relativePathOfDir)) {
-                        StaticResourceInfo.StaticFile newStaticFile = new StaticResourceInfo.StaticFile(dirName.toString(),
-                                attrs.isDirectory(), null);
+                        StaticResourceInfo.StaticFile newStaticFile = new StaticResourceInfo.StaticFile(dirName.toString(), true, null);
                         staticResourceInfo.resourceMap.put(relativePathOfDir, newStaticFile);
 
                         // add newStaticFile in Parent Folder
                         if (staticResourceInfo.resourceMap.containsKey(parentOfDir)) {
```

Comment:
Ah, this needs to be the relative parent path, since that's how you map them. But if you keep a stack of folders you're visiting, you won't need this lookup: you can just take the top of the stack as the current parent.

Reply:
Done